### PR TITLE
glusterfs service: a few fixes and improvements

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

These commits fix issues I found in my production cluster; see their commit messages for individual motivations.

These are the changes I promised in https://github.com/NixOS/nixpkgs/pull/27340#discussion_r139311268.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

FYI @bachp who's another user of this service.

CC @joachifm

### CODE REVIEW

File: nixos/modules/services/network-filesystems/glusterfs.nix [hunk 1]
```diff
@@ -41,6 +41,18 @@ in
         default = "INFO";
       };
 
+      useRpcbind = mkOption {
+        type = types.bool;
+        description = "Enable use of rpcbind. This is required for Gluster's NFS functionality. You may want to turn it off to reduce the attack surface for DDoS reflection attacks. See https://davelozier.com/glusterfs-and-rpcbind-portmap-ddos-reflection-attacks/ and https://bugzilla.redhat.com/show_bug.cgi?id=1426842 for details.";
```
Comment (globin) [inline nixos/modules/services/network-filesystems/glusterfs.nix]: Could you wrap this line?

Comment (nh2) [reply nixos/modules/services/network-filesystems/glusterfs.nix]: @globin Done, also for the other commit.

File: nixos/modules/services/network-filesystems/glusterfs.nix [hunk 2]
```diff
@@ -41,6 +41,53 @@ in
...
+        description = ''
+          The systemd KillMode to use for glusterd.
+
+          glusterd spawns other daemons like gsyncd.
+          If you want these to stop when glusterd is stopped (e.g. to ensure
+          that NixOS config changes are reflected even for these sub-daemons),
+          set this to 'control-group'.
+        '';
+        default = "process";
+      };
```
Comment (bachp) [inline nixos/modules/services/network-filesystems/glusterfs.nix]: Does this need to be configurable? If yes give a quick description in which cases the different modes might be useful.

Comment (nh2) [reply nixos/modules/services/network-filesystems/glusterfs.nix]: @bachp Updated; I have now described both typical use cases.

I have also added a commit that changes the default to `control-group`, as you and me both think that this is the better default (from https://github.com/NixOS/nixpkgs/pull/27340#discussion_r139311268).

**Comment by nh2** [comment]
Copying into main conversation so it doesn't get hidden by `show outdated`:

I have also added a commit that changes the default to `control-group`, as you (@bachp) and me both think that this is the better default (from https://github.com/NixOS/nixpkgs/pull/27340#discussion_r139311268).

**Comment by nh2** [comment]
This is good to go from my side.

Once this is merged, we should switftly go on to #29062, so that we can have a good gluster 3.12 experience on NixOS 17.09.

**Review by bachp — APPROVED** [review]
LGTM

**Pull request merged**